File: KIPOIAPTest_ObjectiveC/KIPOIAPTest_ObjectiveC/ViewController.m
//
//  ViewController.m
//  KIPOIAPTest_ObjectiveC
//
//  Created by Omid Golparvar on 6/30/18.
//  Copyright © 2018 Omid Golparvar. All rights reserved.
//

#import "ViewController.h"
#import "KIPOIAPKit/KIPOIAPKit-Swift.h"

@interface ViewController ()

@end

@interface ViewController () <KipoIAPDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
}

- (IBAction)actionTest:(UIButton *)sender {
	NSLog(@"Starting Test...");
	[KipoIAP SetupWithDelegate: self];
	[KipoIAP PayWithAmount:10000];
}


- (void)kipoCannotPerformWithError:(enum KipoError)error {
	NSLog(@"%@", [KipoIAP ConvertErrorToString:error]);
}

- (void)kipoPaymentWithErrorMessage:(NSString * _Nonnull)errorMessage {
	NSLog(@"%@", errorMessage);
}

- (void)kipoPaymentWithPaymentToken:(NSString * _Nonnull)token {
	
}

@end

File: KIPOIAPTest_Swift/KIPOIAPTest_Swift/ViewController.swift
//
//  ViewController.swift
//  KIPOIAPTest_Swift
//
//  Created by Omid Golparvar on 6/30/18.
//  Copyright © 2018 Omid Golparvar. All rights reserved.
//

import UIKit
import KIPOIAPKit

class ViewController: UIViewController {

	override func viewDidLoad() {
		super.viewDidLoad()
	}
	
	@IBAction func actionTest(_ sender: UIButton) {
		print("Starting Test...")
		KipoIAP.Setup(merchantKey: "09372208177")
		KipoIAP.Setup(delegate: self)
		KipoIAP.Pay(amount: 10000)
		KipoIAP.Check(url)
	}

}

extension ViewController: KipoIAPDelegate {
	
	func kipoCannotPerform(error: KipoError) {
		print(#function + " - Error: " + error.description)
	}
	
	func kipoPayment(paymentToken token: String) {
		print(#function + " - PaymentToken: " + token)
	}
	
	func kipoPayment(errorMessage: String) {
		print(#function + " - Error: " + errorMessage)
	}
	
}
